symengine/sets.h

```diff
@@ -34,7 +31,7 @@ class EmptySet : public Set {
     IMPLEMENT_TYPEID(EMPTYSET)
     //EmptySet(EmptySet const&) = delete;
     void operator=(EmptySet const&) = delete;
-    static RCP<const EmptySet> getInstance();
```

Comment:
`interval` returns `RCP<const Set>`.
`emptyset` returns `RCP<const Set>`
So, I thought it will be better for `getInstance` to return the same, i.e. `RCP<const Set>` to maintain generality.
Futhermore, returning `RCP<const EmptySet>` won't give us any extra help as all the member functions are virtual.